polyply/tests/test_top_parser.py

```diff
@@ -384,6 +384,54 @@ def test_skip_directives(lines):
                                 "gen-pairs":'no',
                                 "fudgeLJ":1.0,
                                 "fudgeQQ":1.0}
+    @staticmethod
+    @pytest.mark.parametrize('lines', (
+         """
+         #define GMXERROR
+         #ifdef GMXERROR
+         #error "This will raise an error."
+         #endif
+         """,
+         """
+         #error This will raise an error
+         """,
+    ))
+    def test_error_parsing(lines):
+        """
+        Test that the #error pragma raises an error.
+        """
+        new_lines = textwrap.dedent(lines)
+        new_lines = new_lines.splitlines()
+        force_field = vermouth.forcefield.ForceField(name='test_ff')
+        top = Topology(force_field, name="test")
+        with pytest.raises(NotImplementedError):
+            polyply.src.top_parser.read_topology(new_lines, top)
```

Comment:
For completion, it is also nice to check that the right error is thrown? In this case, you would also need to make the error messages in lines 396 and 392 consistent. (`#error This will raise an error` vs `#error "This will raise an error."`)

```suggestion
        with pytest.raises(NotImplementedError) as err_info:
            polyply.src.top_parser.read_topology(new_lines, top)
        assert str(err_info.value) == "This will raise an error."
```